js/mapbox-gl-js-common.js

```diff
@@ -2,101 +2,111 @@
 
 var defaultStyle = {};
 
+loadLayers();
+function loadLayers() {
+  $.get('../json/mapLayers.json', function (data) {
+    MAP_LAYERS = data;
+  });
+  $.get('../json/mapLocations.json', function(data) {
+    MAP_LOCATIONS = data;
+  });
+}
+
 // Highlight a layer collection
 function mapHighlight(item) {
-
-  var collectionName = $(item).attr('data-map-layer');
+  var layer = $(item).attr('data-map-layer');
   var color = $(item).attr('data-map-layer-highlight');
 
   // Loop through collection and store defaults before changing them
-  for (var i = 0; i < mapLayerCollection[collectionName].length; i++) {
-
-    var obj = mapLayerCollection[collectionName][i];
-
-    // Choose an appropriate property to change
-    if (map.getLayer(obj).type == 'raster')
-      prop = 'raster-opacity';
-    if (map.getLayer(obj).type == 'fill')
-      prop = 'fill-color';
-    if (map.getLayer(obj).type == 'line')
-      prop = 'line-color';
-    if (map.getLayer(obj).type == 'circle')
-      prop = 'circle-color';
-
+  MAP_LAYERS[layer].forEach(function(mapLayer) {
     var propObj = {};
-    propObj[prop] = map.getPaintProperty(obj, prop);
 
-    defaultStyle[obj] = propObj;
-    map.setPaintProperty(obj, prop, color);
+    switch(map.getLayer(mapLayer).type) {
+      case 'raster':
+        prop = 'raster-opacity';
```

Comment:
You probably want to do `var prop` at the top of the function or so, else `prop` is going to become accidentally global, and I'm not sure that's the intention.

Reply:
Just an addendum: if the intention IS to make `prop` global, I prefer doing so explicitly by doing `window.prop` somewhere so that a reader of the code knows the global is intentional and not an accident.